### ISSUE TITLE
Site Administrator or manage multiple sites with same user

### ISSUE THREAD

**Issue by lesion** [madoublet/respond#519]
Hey,
I'm trying respond6, it seems really good.
I don't know if it's the right place to ask questions but I'll try.
I'm was searching a way to have one global __Site Admin__ that could manage all sites,
Searching for it I found this: http://r5.respondcms.com/documentation/multiple-sites
but it seems to me that this functionality is not present in R6 anymore, it's correct?
(I also found [this](https://groups.google.com/forum/#!topic/respond-cms/KwbBuoMZw0g) answer on google groups but cannot find this SITE_ADMIN constant anywhere in R6.)
Is downgrading to Respond5 a good solution to get that feature?
Another idea was to add to each sites the admin user:
I see that R6 can have multiple users per sites, but is it possible to login with one user (with permissions for more sites) and switch the sites to administrate without a new login with same credential? maybe with LDAP could we achieve that?

many thanks

**Comment by axis80** [comment]
You're correct - in R6 there is no global admin user who can manage all sites, not is there any way that a single user can switch between managing different sites.  All of the sites are siloed, if I'm using that term correctly.

You may be correct that it's possible to achieve what you're after using LDAP, assuming you already have an LDAP server running.  With some additional code, Oauth2 or SAML might also be made to work.

If you can share more about your use case it may help us to shape development moving forward.  What is it that you are trying to do that you can't?  Manage content on multiple sites?  Mange user accounts across multiple sites?  Delete sites?

FWIW, I maintain a couple dozen Respond sites that are all hosted as part of the same installation.  I use LastPass to store the email address and password for each one.  I log into different sites to update content all day long.  That setup works pretty well for me.  Do you think it would work well for you?

**Comment by lesion** [comment]
> If you can share more about your use case it may help us to shape development moving forward. What is it that you are trying to do that you can't? Manage content on multiple sites? Mange user accounts across multiple sites? Delete sites?

My needs are:
- a sort of admin panel (to see all hosted sites without using the cli :) (after the login of course)
- a way to create sites from the admin panel itself
- a way to go through sites settings using one login only

I can also help with this if you suggest me the right path.

FIY I found a critical path traversal issue during site creation. this could lead to getting other sites credential (hashed of course) and create a DOS (full disk). should I open another issue here or you prefer another place for this kind of report?

**Comment by madoublet** [comment]
Thanks @lesion.  Could you email me the issue directly (matt@matthewsmith.com) along with any recommendations you have to fix it?

I can see the need for an admin tool.  I purposely left it out of R6, because I think it might make sense as a standalone app.

**Comment by axis80** [comment]
Can you please CC me on that as well, at derrick.miller@axis80.com?

**Comment by lesion** [comment]
> I can see the need for an admin tool. I purposely left it out of R6, because I think it might make sense as a standalone app.

very nice idea, did you already started with this ?
btw. there's some documentation on LDAP authentication? do you think this could solve my issue with admin authentication?

thanks

**Comment by lidio601** [comment]
+1

**Comment by madoublet** [comment]
I have not started it.  I did not work on the LDAP piece.  It was PR created by a user with a corporate network.  I can't think of any easy way to do what you want right now.  I plan to do some work around the login to get rid of the /login/site path.  So, perhaps, there could be a solution in that work.
